src/vdaf/prio3.rs

```diff
@@ -779,23 +779,105 @@ impl HelperShare {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::vdaf::{run_vdaf, run_vdaf_prepare};
 
     #[test]
     fn test_prio3_count64() {
-        let prio3 = Prio3Count64::new(Suite::Blake3, 23).unwrap();
+        let prio3 = Prio3Count64::new(Suite::Blake3, 2).unwrap();
+
+        assert_eq!(
+            run_vdaf(&prio3, &(), [1, 0, 0, 1, 1]).unwrap(),
+            Prio3Result(3)
+        );
+
         let (_, verify_params) = prio3.setup().unwrap();
         let nonce = b"This is a good nonce.";
+
+        let input_shares = prio3.shard(&(), &0).unwrap();
+        run_vdaf_prepare(&prio3, &verify_params, &(), nonce, input_shares).unwrap();
+
         let input_shares = prio3.shard(&(), &1).unwrap();
-        eval_vdaf(
-            &prio3,
-            &input_shares,
-            &verify_params,
-            nonce,
-            Some(Prio3Result(1)),
-        )
-        .unwrap();
-
-        // TODO(cjpatton) Add failure test cases.
+        run_vdaf_prepare(&prio3, &verify_params, &(), nonce, input_shares).unwrap();
+    }
+
+    #[test]
+    fn test_prio3_sum64() {
+        let prio3 = Prio3Sum64::new(Suite::Blake3, 3, 16).unwrap();
+
+        assert_eq!(
+            run_vdaf(&prio3, &(), [0, (1 << 16) - 1, 0, 1, 1]).unwrap(),
+            Prio3Result((1 << 16) + 1)
+        );
+
+        let (_, verify_params) = prio3.setup().unwrap();
+        let nonce = b"This is a good nonce.";
+
+        // TODO Check that the expected error is returned (requires `VdafError` to implement `Eq`).
```

Comment:
If what you want to do is check that you got the expected variant of `VdafError`, then you don't need to compare by equality. You can use [`assert_matches!`](https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html) as in [`test_accumulate`](https://github.com/abetterinternet/libprio-rs/blob/fca3f26f572ede6e6f5f778d797661f1dbdb2566/src/field.rs#L660).